spec/services/flags/create_flag_spec.rb

```diff
@@ -1,105 +1,78 @@
 # frozen_string_literal: true
 
 RSpec.describe(Flags::CreateFlag) do
-  subject(:result) do
-    described_class.call(
-      guardian: current_user.guardian,
-      name: name,
-      description: description,
-      applies_to: applies_to,
-      require_message: require_message,
-      enabled: enabled,
-    )
+  describe described_class::Contract, type: :model do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:description) }
+    it { is_expected.to validate_length_of(:name).is_at_most(Flag::MAX_NAME_LENGTH) }
+    it { is_expected.to validate_length_of(:description).is_at_most(Flag::MAX_DESCRIPTION_LENGTH) }
+    it { is_expected.to validate_inclusion_of(:applies_to).in_array(Flag.valid_applies_to_types) }
   end
 
-  let(:name) { "custom flag name" }
-  let(:description) { "custom flag description" }
-  let(:applies_to) { ["Topic"] }
-  let(:enabled) { true }
-  let(:require_message) { true }
+  describe ".call" do
+    subject(:result) { described_class.call(**params, **dependencies) }
```

Comment:
As this PR has been extracted from https://github.com/discourse/discourse/pull/29129, I just changed `params:` to `**params` :slightly_smiling_face: